### PR TITLE
8.6.0 by removing patches

### PR DESCRIPTION
**Pull request by aulemahal**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged) (the bot did it)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This is a test to see if the builds pass when some patches are removed.

### CODE REVIEW

**Comment by aulemahal** [comment]
Well, the nompi builds on Linux are failing as `gcc` can't be found. Weird, but should be easily fixed by someone who knows this stuff ?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -57,8 +57,8 @@ requirements:
     - netcdf-fortran * {{ mpi_prefix }}_*  # [build_platform != target_platform and mpi == 'openmpi']
     - hdf5 * {{ mpi_prefix }}_*  # [build_platform != target_platform and mpi == 'openmpi']
     - libnetcdf * {{ mpi_prefix }}_*  # [build_platform != target_platform and mpi == 'openmpi']
+    - {{ compiler('c') }}  # [not win]
     # Unix
-    - {{ compiler('c') }}  # [unix]
```
Comment (xylar) [inline recipe/meta.yaml:61]: @aulemahal unfortunately, this won't change anything.  `unix` is the same as `not win`.

**Comment by aulemahal** [comment]
Woups, I'll revert that..

Otherwise, the Windows build is failing and it looks it because of this issue : https://github.com/esmf-org/esmf/issues/195
The discussion there says to remove the option, but the build rules in the ESMF repo mention that this is needed on windows.
https://github.com/esmf-org/esmf/blob/977d04e9eab2aa3f0f301ee66fbcf5b6c54cb272/build_config/MinGW.gfortran.default/build_rules.mk#L143

**Comment by xylar** [comment]
@aulemahal. feel free to make a patch to fix that issue.  You can google around and look at some examples to see how patches are made but my process is:
1. download the ESMF source code we use here
2. untar it once and rename the resulting directory to add `-patch` at the end
3. untar it again
4. modify the code as needed in the version with the `-patch` 
5. run `diff -ruN esmf esmf-patch > recipe/000X-reason-for-fix.patch`
6. add the patch in the `recipe/meta.yaml`

Happy to look over what you come up with.

**Comment by xylar** [comment]
In the previous CI, we're seeing:
```
gcc -c -std=c99 -O -DNDEBUG -fPIC -fopenmp -I$SRC_DIR/src/Infrastructure/stubs/mpiuni -I$SRC_DIR/src/Infrastructure/stubs/mpiuni/../include -I$SRC_DIR/build_config/Linux.gfortran.default -I$SRC_DIR/src/Infrastructure -I$SRC_DIR/src/Superstructure -I$SRC_DIR/src/include   -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_MPIUNI -I$SRC_DIR/src/Infrastructure/stubs/mpiuni -DESMF_LOWERCASE_SINGLEUNDERSCORE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_LAPACK_INTERNAL=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_NETCDF=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=mpiuni -DESMF_DIR=$SRC_DIR -D__SDIR__='"src/Infrastructure/stubs/mpiuni"' -DESMF_CSTD=99 $SRC_DIR/src/Infrastructure/stubs/mpiuni/mpi.c -o $SRC_DIR/obj/objO/Linux.gfortran.64.mpiuni.default/mpi.o
make[8]: gcc: No such file or directory
```
This is presumably because of a file where the compiler is hard-coded to `gcc` instead of being general.  For example, here it will be something like `$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gcc`.  @aulemahal, maybe you can track down where that mistake is happening?

**Comment by xylar** [comment]
Sorry, what I said about `gcc` is just incorrect.  The solution is https://github.com/conda-forge/esmf-feedstock/pull/110/commits/8b266c9ef4b42e078aa28bf6f533cf2e199173ea

**Comment by xylar** [comment]
@aulemahal, thank you for working on this, it helped me quite a bit!  I'm closing this in favor of #110.

**Pull request closed without merging**